### PR TITLE
[CodeCompletion] Suggest `in` keyword in `for … in` loop

### LINKED ISSUE #58633: Suggest `in` keyword in `for … in` loop
In the following new test case we don’t suggest the `in` keyword although it’s valid.

```swift
// RUN: %swift-ide-test -code-completion -source-filename %s -code-completion-token COMPLETE

func test(inValues: [Int]) {
  for value #^COMPLETE^# 
}
```

rdar://92653535

### PR DESCRIPTION
**Pull request by simanerush**
Resolves #58633.

### CODE REVIEW

**Comment by ahoppen** [comment]
@swift-ci Please smoke test

**Review by ahoppen — APPROVED** [review]
Thanks for your contribution to the Swift compiler. This is looking great 👍 

If you want to mark this as ready for review, I think it’s ready to be merged.

**Comment by simanerush** [comment]
@ahoppen I just ran `clang-format` on these changes and marked as ready for review. Thank you for quick feedback!

**Comment by AnthonyLatsis** [comment]
@swift-ci please smoke test

File: test/IDE/complete_loop.swift [hunk 1]
```diff
@@ -5,6 +5,8 @@
 // RUN: %target-swift-ide-test -code-completion -source-filename %s -code-completion-token=LOOP_3 | %FileCheck %s -check-prefix=LOOP_3
 // RUN: %target-swift-ide-test -code-completion -source-filename %s -code-completion-token=LOOP_4 | %FileCheck %s -check-prefix=LOOP_4
 // RUN: %target-swift-ide-test -code-completion -source-filename %s -code-completion-token=LOOP_5 | %FileCheck %s -check-prefix=LOOP_5
+// RUN: %target-swift-ide-test -code-completion -source-filename %s -code-completion-token=LOOP_6 | %FileCheck %s -check-prefix=LOOP_6
+// RUN: %target-swift-ide-test -code-completion -source-filename %s -code-completion-token=LOOP_7 | %FileCheck %s -check-prefix=LOOP_6
```
Comment (LucianoPAlmeida) [inline test/IDE/complete_loop.swift:9]: Is this correct? Maybe I'm not fully understanding the intention here but I would expect `LOOP_7` in this second run
```suggestion
// RUN: %target-swift-ide-test -code-completion -source-filename %s -code-completion-token=LOOP_7 | %FileCheck %s -check-prefix=LOOP_7
```

File: test/IDE/complete_loop.swift [hunk 2]
```diff
@@ -67,3 +69,14 @@ class C {
...
+// https://github.com/apple/swift/issues/58633
+do {
+  for value #^LOOP_6^#
+}
+do {
+  for value #^LOOP_7^# 1..<7 {}
+}
+// LOOP_6: Begin completions, 1 items
+// LOOP_6-CHECK-NEXT: Keyword[in]/None:                   in; name=in
+// LOOP_6-CHECK-NEXT: End completions
```
Comment (LucianoPAlmeida) [inline test/IDE/complete_loop.swift:82]: @ahoppen Should we also check for `LOOP_7` completions?

File: test/IDE/complete_loop.swift (hunk 1 above)
Comment (AnthonyLatsis) [reply test/IDE/complete_loop.swift:9]: The FileCheck prefix is reused because we are expecting the same output from both test cases.

Comment (LucianoPAlmeida) [reply test/IDE/complete_loop.swift:9]: Ah I see, thanks @AnthonyLatsis

**Pull request merged**